playbooks/provisioning/openstack/post-provision-openstack.yml

```diff
@@ -45,6 +45,14 @@
   become: False
   roles:
     - dns-server-detect
+  tasks:
```

Comment:
the dns-server-detect role always sets the `private_dns_server` to `external_nsupdate_keys.private.server`, while for this case we want it to take the 1st dns server private IP (from the Heat stack)